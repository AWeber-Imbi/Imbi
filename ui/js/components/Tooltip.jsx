import React, { useState } from "react";

import { Tooltip } from "reactstrap";

export default function(props) {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <Tooltip
      {...props}
      autohide={false}
      isOpen={isOpen}
      delay={0}
      toggle={() => {
        setIsOpen(!isOpen);
      }}
      data-testid="tooltip-display"
    >
      {props.children}
    </Tooltip>
  );
}
