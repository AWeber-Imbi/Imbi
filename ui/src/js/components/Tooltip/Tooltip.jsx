import PropTypes from "prop-types";
import React, {useState} from 'react'

function Tooltip({children, value}) {
  const [showTooltip, setShowTooltip] = useState(false);
  return (
    <>
      <div className="flex flex-1"
           onMouseOver={() => setShowTooltip(true)}
           onMouseOut={() => setShowTooltip(false)}>
        {children}
      </div>
      <div className={showTooltip === true ? 'visible' : 'hidden'}>
        <div className="absolute z-50 pt-9 -ml-8">
          <svg className="text-black bottom-full" x="0px" y="0px" height="10px" width="10px"
               xmlSpace="preserve">
            <polygon className="fill-current" points="0,10 5,5 10,10"/>
          </svg>
          <div className="bg-black z-50 text-sm -ml-3 rounded text-white p-1 pl-2 pr-2 ">
            {value}
          </div>
        </div>
      </div>
    </>
  )
}

Tooltip.propTypes = {
  children: PropTypes.any,
  value: PropTypes.string.isRequired
}

export default Tooltip
