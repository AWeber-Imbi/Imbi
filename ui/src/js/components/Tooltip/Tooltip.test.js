import React from "react"
import {render} from "@testing-library/react"
import {configure, shallow} from "enzyme"
import "@testing-library/jest-dom/extend-expect"
import Adapter from "enzyme-adapter-react-16"

import Tooltip from "./Tooltip"

configure({adapter: new Adapter()})

describe("Tooltip", () => {
  it("should render the hidden tooltip Tooltip", () => {
    const tree = render(
      <Tooltip value="Tooltip">
        <div id="child">Foo</div>
      </Tooltip>)
    expect(tree).toMatchSnapshot()
  })

  it("should change the visibility on mouse over/out", () => {
    const element = shallow(
      <Tooltip value="Tooltip">
        <div id="child">Foo</div>
      </Tooltip>)

    let divs = element.find("div")
    expect(divs.at(2).hasClass("hidden")).toBe(true)
    expect(divs.at(2).hasClass("visible")).toBe(false)
    divs.at(0).simulate("mouseover")
    element.rerender()
    divs = element.find("div")
    expect(divs.at(2).hasClass("hidden")).toBe(false)
    expect(divs.at(2).hasClass("visible")).toBe(true)
    divs.at(0).simulate("mouseout")
    element.rerender()
    divs = element.find("div")
    expect(divs.at(2).hasClass("hidden")).toBe(true)
    expect(divs.at(2).hasClass("visible")).toBe(false)
  })
})
