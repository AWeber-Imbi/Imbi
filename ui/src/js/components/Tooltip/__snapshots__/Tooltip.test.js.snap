// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render the hidden tooltip Tooltip 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-1"
      >
        <div
          id="child"
        >
          Foo
        </div>
      </div>
      <div
        class="hidden"
      >
        <div
          class="absolute z-50 pt-9 -ml-8"
        >
          <svg
            class="text-black bottom-full"
            height="10px"
            width="10px"
            x="0px"
            xml:space="preserve"
            y="0px"
          >
            <polygon
              class="fill-current"
              points="0,10 5,5 10,10"
            />
          </svg>
          <div
            class="bg-black z-50 text-sm -ml-3 rounded text-white p-1 pl-2 pr-2 "
          >
            Tooltip
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-1"
    >
      <div
        id="child"
      >
        Foo
      </div>
    </div>
    <div
      class="hidden"
    >
      <div
        class="absolute z-50 pt-9 -ml-8"
      >
        <svg
          class="text-black bottom-full"
          height="10px"
          width="10px"
          x="0px"
          xml:space="preserve"
          y="0px"
        >
          <polygon
            class="fill-current"
            points="0,10 5,5 10,10"
          />
        </svg>
        <div
          class="bg-black z-50 text-sm -ml-3 rounded text-white p-1 pl-2 pr-2 "
        >
          Tooltip
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
